src/components/dashboard/FeedItems/FeedListItem.js

```diff
@@ -37,6 +39,7 @@ const FeedListItem = (props: FeedListItemProps) => {
   }
   const disableAnimForTests = Config.env === 'test'
   const easing = 'ease-in'
+  const onPressAction = item.action ? eval(item.action) : undefined
```

Comment:
```javascript
import { isFunction } from  'lodash'

const { id, action } = item 
const onItemPress = useCallback(() => {
  if (!action) {
    onPress(id);
    return
  }

  const [method, argsList] = action.split(':')

  if (isFunction(navigation[method])) {
    navigation[method](...argsList.split(','))
  }
}, [id, action, navigation])

```

Reply:
**eval() is EVIL** we shouldn't use it never
if we need to have some string-based declaration what action should to do -
we could add simple (but formalized) macro-like command format

in example i proposed it's:
```
<navigation's method to call>:<arg1>[,<arg2>,...]
```

then we parse action string, validating it (i skipped this in my example but of course we have to add at least `['allowed', 'methods', 'to', 'call'].includes(method)` check)
and interpreting